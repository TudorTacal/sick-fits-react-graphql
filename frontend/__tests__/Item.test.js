import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Item from '../components/Item';

const fakeItem = {
  id: 'ABC123',
  title: 'A Cool Item',
  price: 5000,
  description: 'This item is really cool!',
  image: 'dog.jpg',
  largeImage: 'largedog.jpg',
};

describe('<Item />', () => {
  it('renders and displays properly', () => {
    const wrapper = shallow(<Item item={fakeItem} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
