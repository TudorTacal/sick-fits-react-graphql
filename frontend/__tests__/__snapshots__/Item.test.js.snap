// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> renders and displays properly 1`] = `
<ItemStyles__Item>
  <img
    alt="A Cool Item"
    src="dog.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <a>
        A Cool Item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $50
  </PriceTag>
  <p>
    This item is really cool!
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "/update",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <a>
        Edit ✏️
      </a>
    </Link>
    <AddToCart
      id="ABC123"
      image="dog.jpg"
      price={5000}
      title="A Cool Item"
    />
    <DeleteItem
      id="ABC123"
    >
      Delete this item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
